kyo-prelude/shared/src/main/scala/kyo/Layer.scala

```diff
@@ -288,34 +287,30 @@ object Layer:
         kyo.internal.LayerMacros.make[Target](layers*)
 
     private[kyo] object internal:
-        case class And[Out1, Out2, S1, S2](lhs: Layer[Out1, S1], rhs: Layer[Out2, S2])                   extends Layer[Out1 & Out2, S1 & S2]
-        case class To[Out1, Out2, S1, S2](lhs: Layer[?, ?], rhs: Layer[?, ?])                            extends Layer[Out1 & Out2, S1 & S2]
-        case class FromKyo[In, Out, S](kyo: () => TypeMap[Out] < (Env[In] & S))(using val tag: Tag[Out]) extends Layer[Out, S]
+        case class And[Out1, Out2, S1, S2](first: Layer[Out1, S1], second: Layer[Out2, S2 & Env[Out1]])  extends Layer[Out1 & Out2, S1 & S2]
+        case class To[Out1, Out2, S1, S2](first: Layer[Out1, S1], second: Layer[Out2, S2 & Env[Out1]])   extends Layer[Out2, S1 & S2]
```

Comment:
@hearnadam the main problem was here. The previous definition of To was ditching other Envs from the type definition.